### PR TITLE
Mobile home page

### PR DESCRIPTION
**Pull request by glen-aot**
# Description

Make static homepage responsive, updates to html and css files. Also updates images to webp

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1347-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1347-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1347-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1347-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1347-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1347-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1347-dops.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1347-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/package-lock.json [hunk 1]
```diff
@@ -73,7 +73,7 @@
         "vitest-canvas-mock": "^0.3.3"
       }
     },
-    "../../../../../../latest": {
+    "../../../../../latest": {
```
Comment (krishnan-aot) [inline frontend/package-lock.json]: is this change needed?

File: frontend/static_home_page/css/styles.css [hunk 2]
```diff
@@ -1,44 +1,51 @@
...
     font-family: 'BCSans';
+    font-display: swap;
 }
 
 
 :root {
-
     /* Colors: */
-    --focus-blue: #3B99FC;
+    --focus-blue: #0000FF;
```
Comment (krishnan-aot) [inline frontend/static_home_page/css/styles.css]: I don't think the new color is the right one. You can refer to our current theme colors here: [Refer](https://github.com/bcgov/onroutebc/blob/main/frontend/src/themes/bcGovStyles.tsx)

File: frontend/static_home_page/css/styles.css [hunk 3]
```diff
@@ -35,10 +40,12 @@
...
     --bc-primary-blue: #003366;
+    --bc-secondary-blue: #133262;
     --bc-border-grey: #DBDCDC;
     --bc-text-links-blue: #1A5A96;
     --bc-background-light-grey: #F2F2F2;
     --bc-background-blue-grey: #F4F5F8;
+    --bc-background-blue: #455A86;
     
     /* Font/text values */
     --unnamed-font-family-bc-sans: 'BCSans';
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:51]: Is there a reason why you need to add "--unnamed-" prefix to these variables?

File: frontend/static_home_page/css/styles.css [hunk 4]
```diff
@@ -93,28 +111,40 @@ p {
     letter-spacing: var(--unnamed-character-spacing-0);
 }
 
-h4 {
+h5, .h5 {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:114]: Why do we need the class name of ".h5" when we already have h5 element? Similarly for h4 and h3 below as well? Can we replace the ".h5", ".h4", and ".h3" with their html counterparts instead? (ie. h5, h4, h3)

File: frontend/static_home_page/css/styles.css [hunk 5]
```diff
@@ -83,8 +98,11 @@ h2 {
...
-    margin-block-start : 0.5em;
-    margin-block-end : 0.5em;
+    margin-bottom: 16px;
+}
+
+p:last-child {
+    margin-bottom: 0;
 }
 
 .titleText {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:108]: Personally when it comes to conventions, I prefer ".title-text" over ".titleText" for class names, but I guess this isn't a priority right now (as there will be a lot of changes to be made)

File: frontend/static_home_page/css/styles.css [hunk 6]
```diff
@@ -127,191 +157,325 @@ h3 {
...
-    height: 1796px;
-    background: #F1F1F1 0% 0% no-repeat padding-box;
-    opacity: 1;
+/* Utility Classes*/
+.margin-none {
+    margin: 0;
+}
+
+.border-none {
+    border-width: 0px !important;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:166]: Try to avoid using "!important" in any of the CSS styles. There should be other, better ways of doing this.

File: frontend/static_home_page/css/styles.css [hunk 7]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+}
+
+body {
+    margin: 0;
+    background-color: #F1F1F1;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:171]: Is it possible to add this color to the list of variables above, and reuse the variable here instead?

File: frontend/static_home_page/css/styles.css [hunk 8]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+    padding-bottom: 80px;
+}
+
+.container {
+    max-width: calc(1470px + (16px * 2));
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:176]: Is it possible to add any of these measurements to the list of variables at the top? (eg. 1470px or something else)? Also why do you write this as 16px * 2 instead of simply 32px (which could also be used as a variable instead)?

File: frontend/static_home_page/css/styles.css [hunk 9]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+    margin: 0 auto;
+    padding: 0 var(--padding-16);
 }
 
 .homeheader {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:181]: Be consistent with the class name conventions (ie. either ".home-header" or ".homeHeader")

File: frontend/static_home_page/css/styles.css [hunk 10]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+        width: 208px;
+    }
 }
 
 .homeBClogo {
     top: 0;
     left: 0;
-    width: 1920px;
+    width: 100%;
     height: 3444px;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:203]: Consider adding this measurement to list of variables above

File: frontend/static_home_page/css/styles.css [hunk 11]
```diff
@@ -127,191 +157,325 @@ h3 {
...
     background: transparent url('../img/Home - Province of British Columbia.png') 0% 0% no-repeat padding-box;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:204]: Is it possible to rename the png file name, it's not ideal to have spacings within filenames.

File: frontend/static_home_page/css/styles.css [hunk 12]
```diff
@@ -127,191 +157,325 @@ h3 {
...
     opacity: 1;
 }
 
-.homeTruckImage {
-    position: absolute;
-    top: 89px;
-    width: 1920px;
+.heroImage {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:208]: If there's no styling for this class (empty or commented out), then remove it altogether

File: frontend/static_home_page/css/styles.css [hunk 13]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+    /* display: block; */
+}
+
+.heroImage img {
+    width: 100vw;
+    height: 580px;
+    object-fit: cover;
+}
+
+/* .homeTruckImage {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:218]: Remove this if it's not being used anymore

File: frontend/static_home_page/css/styles.css [hunk 14]
```diff
@@ -127,191 +157,325 @@ h3 {
...
-    position: absolute;
-    top: 89px;
-    width: 1920px;
+.heroImage {
+    /* display: block; */
+}
+
+.heroImage img {
+    width: 100vw;
+    height: 580px;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:214]: Consider moving this to list of variables?

File: frontend/static_home_page/css/styles.css [hunk 15]
```diff
@@ -127,191 +157,325 @@ h3 {
...
 }
 
 .homeHeaderLogo {
-    position: relative;
-    top: 15px;
-    left: 180px;
-    width: 400px;
-    height: 89px;
-    background: transparent url('../img/BC_Header.png') 0% 0% no-repeat padding-box;
+    width: 115px;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:189]: Consider adding this to list of variables at the top

File: frontend/static_home_page/css/styles.css [hunk 16]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+    transform: translateY(-8px);
+}
+
+@media screen and (min-width: 768px) {
+    .homeHeaderLogo {
+        width: 208px;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:195]: Consider adding this to list of variables at the top as well, just distinguish the variable name from the normal home header logo width variable name

File: frontend/static_home_page/css/styles.css [hunk 17]
```diff
@@ -127,191 +157,325 @@ h3 {
...
 
 .homeTruckImageFade {
     position: absolute;
     top: 458px;
-    width: 1920px;
+    width: 100%;
     height: 340px;
-    background: transparent linear-gradient(180deg, #F1F1F100 0%, #F1F1F1 100%) 0% 0% no-repeat padding-box;
-    opacity: 1;
+    background: transparent linear-gradient(180deg, #F1F1F100 0%, #F1F1F1 50%) 0% 0% no-repeat padding-box;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:230]: Definitely consider adding this to list of variables, either in part (f1f1f100 or f1f1f1), or the whole thing (ie. transparent ... padding-box)

File: frontend/static_home_page/css/styles.css [hunk 18]
```diff
@@ -127,191 +157,325 @@ h3 {
...
     opacity: 1;
-}
+} */
 
 .homeTruckImageFade {
     position: absolute;
     top: 458px;
-    width: 1920px;
+    width: 100%;
     height: 340px;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:229]: Consider adding this to list of variables

File: frontend/static_home_page/css/styles.css [hunk 19]
```diff
@@ -127,191 +157,325 @@ h3 {
...
-    background: transparent linear-gradient(180deg, #F1F1F100 0%, #F1F1F1 100%) 0% 0% no-repeat padding-box;
-    opacity: 1;
+    background: transparent linear-gradient(180deg, #F1F1F100 0%, #F1F1F1 50%) 0% 0% no-repeat padding-box;
 }
 
-.homeTitleBox {
+.verticalContainer {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:233]: There should probably be a better name for this class, as "verticalContainer" doesn't really tell much about what exactly it's describing

File: frontend/static_home_page/css/styles.css [hunk 20]
```diff
@@ -127,191 +157,325 @@ h3 {
...
     position: absolute;
-    top: 185px;
-    left: 235px;
-    width: 680px;
-    height: 218px;
+    padding-top: 90px ;
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:235]: Consider adding these independent measurements (eg. ...px or specific colours like 1f3564) to list of variables at the top, and reuse them wherever necessary. This goes for all of the measurements below.

File: frontend/static_home_page/css/styles.css [hunk 21]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+    top: 96px;
+    width: 100%;
+}
+
+.homeTitleBox {
+    max-width: 680px;
     background: #1F3564 0% 0% no-repeat padding-box;
     opacity: 0.8;
     color: var(--white);
     font: var(--unnamed-font-style-normal) normal var(--unnamed-font-weight-normal)
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:245]: This looks a bit strange, consider creating a variable called --home-title-box-font (or something like that) and assign to its value this line

File: frontend/static_home_page/css/styles.css [hunk 22]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+        font-size: var(--unnamed-font-size-36)
+    }
+}
+
+.homeTitleBox .titleText {
+    margin-top: 8px;
+    margin-bottom: 56px;
 }
 
 .homeContent2Col {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:269]: Consider finding a better name for this class

File: frontend/static_home_page/css/styles.css [hunk 23]
```diff
@@ -127,191 +157,325 @@ h3 {
...
-    padding : 10px 20px 10px 20px;
+    padding : var(--padding-24) var(--padding-16);
 }
 
+.homeContentBoxContent p {
+    margin-top: 0;
+}
+
+
 .homeContentBoxContent2Col {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:308]: Consider finding a better name for this class

File: frontend/static_home_page/css/styles.css [hunk 24]
```diff
@@ -127,191 +157,325 @@ h3 {
...
-    position: relative;
-    top: 0;
+    border-top : 1px solid var(--bc-border-grey);
     display: flex;
+    flex-direction: column;
+}
+
+.homeContentBoxContent2Col .homeBorderedContentCol:not(:last-child) {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:314]: The last class name looks suspiciously complicated. Perhaps consider adding a unique class name for the last element (the one referred to by last-child) and apply the styling to that class name?

File: frontend/static_home_page/css/styles.css [hunk 25]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+
+.homeBorderedContentCol {
+    padding: var(--padding-16) var(--padding-16) var(--padding-24);
+}
+
+.homeContentCol {
+   width: 100%;
 }
 
 .homeContentCol1 {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:341]: Consider finding a better class name for col1 and col2, as it's difficult to infer what those are.

File: frontend/static_home_page/css/styles.css [hunk 26]
```diff
@@ -127,191 +157,325 @@ h3 {
...
     display: block;
     text-align: center;
     opacity: 1;
     border-radius: 4px;
     font-family: var(--unnamed-font-family-bc-sans);
+    border: 4px solid transparent;
+    width: fit-content;
+}
+
+.btnSm {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:389]: For descriptive class names like these (types of buttons), you can use BEM conventions (ie. for this small button, use ".btn--sm", and for blue buttons you can use ".btn--blue")

File: frontend/static_home_page/css/styles.css [hunk 27]
```diff
@@ -127,191 +157,325 @@ h3 {
...
 
 .link {
     color : var(--bc-text-links-blue);
 }
 
+.link:hover, .link:focus {
+    color: var(--focus-blue);
+}
+
 .homeContentBoxContent3Col {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:436]: Once again, not sure what 3Col refers to, so perhaps consider finding a better class name here

File: frontend/static_home_page/css/styles.css [hunk 28]
```diff
@@ -127,191 +157,325 @@ h3 {
...
+        flex-direction: row;
+    }
+
+    .homeContentBoxContent3Col .homeBorderedContentCol:not(:last-child) {
+        border-bottom: none;
+        border-right:  1px solid var(--bc-border-grey);
+    }
+}
+
+#permitCentre .homeContentBoxContent2Col  {
```
Comment (zgong-gov) [inline frontend/static_home_page/css/styles.css:458]: Avoid using ids, and try to use class names instead whenever possible

File: frontend/static_home_page/home.html [hunk 29]
```diff
@@ -1,151 +1,340 @@
...
+                </div>
+                <div class="homeContentBoxContent">
+                  <h3>Contact Us</h3>
+                  <p>
+                    If you need assistance, or have questions about processing
+                    your permit, you can contact the Provincial Permit Centre.
+                  </p>
+                  <div class="homeContentBoxContent2Col border-none">
+                    <div class="homeContentCol">
+                      <h4 class="h5 margin-none">Phone</h4>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:205]: This doesn't make sense, assigning class name ".h5" to an element of h4. Either change the element to h5 instead, or remove the class name.

File: frontend/static_home_page/home.html [hunk 30]
```diff
@@ -1,151 +1,340 @@
...
+                      <p>
+                        <a href="tel:+18005599688" class="link"
+                          >1-800-559-9688</a
+                        >
+                      </p>
+                      <h4 class="h5 margin-none">Email</h4>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:211]: Similarly here (ie. remove the ".h5" class name, or change the element to h5 element), whichever one has the appropriate style in the css.

File: frontend/static_home_page/home.html [hunk 31]
```diff
@@ -1,151 +1,340 @@
...
+                      credit card only. Over time onRouteBC will offer more
+                      permit types and payment methods. During transition,
+                      permits that are not available in onRouteBC continue to be
+                      available from the existing system by contacting the
+                      Provincial Permit Center.
+                    </p>
+                  </div>
+                  <div class="homeContentBoxContent2Col">
+                    <div class="homeBorderedContentCol">
+                      <h3 class="h4">Term Oversize Permits Only</h3>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:121]: This doesn't make sense, assigning class name of ".h4" to an element of h3. Remove the class name, or change the element to h4, whichever one preserves the styling.

File: frontend/static_home_page/home.html [hunk 32]
```diff
@@ -1,151 +1,340 @@
...
+                      <a
+                        role="button"
+                        class="btn btnBlue"
+                        href="/?idpHint=bceidboth"
+                      >
+                        Log into onRouteBC
+                      </a>
+                    </div>
+                    <div class="homeBorderedContentCol">
+                      <h3 class="h4">Other permit types and payment methods</h3>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:135]: Same here.

File: frontend/static_home_page/home.html [hunk 33]
```diff
@@ -1,151 +1,340 @@
...
+                  </div>
+                </div>
+              </section>
+              <section class="homeContentBox" id="permitInformation">
+                <div class="homeContentBoxHeader">
+                  <h2>Permit information</h2>
+                </div>
+                <div class="homeContentBoxContent3Col">
+                  <div class="homeBorderedContentCol">
+                    <h3 class="h4">Commercial Transport Procedures</h3>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:160]: And here.

File: frontend/static_home_page/home.html [hunk 34]
```diff
@@ -1,151 +1,340 @@
...
+                    <p>
+                      Legislation, policies, standards and guidelines to be
+                      followed in the operation of a commercial transportation
+                      business in British Columbia.
+                    </p>
+                  </div>
+                  <div class="homeBorderedContentCol">
+                    <h3 class="h4">Out of Province Permits</h3>
```
Comment (zgong-gov) [inline frontend/static_home_page/home.html:168]: And here.

File: frontend/static_home_page/css/styles.css (hunk 5 above)
Comment (zgong-gov) [reply frontend/static_home_page/css/styles.css:108]: ie. use kebab case (dashes "-") and all lower case over camel case (titleText) for class names. For js variables however, use camel case instead, and for typescript types and interfaces use Pascal case (eg. SomeType), and for constants use snake case with all capitals (eg. SOME_CONSTANT_HERE)

File: frontend/static_home_page/css/styles.css (hunk 3 above)
Comment (glen-aot) [reply frontend/static_home_page/css/styles.css:51]: This is the convention that was already in place when I took on this task, trying to stay consistent with the code already written

File: frontend/static_home_page/css/styles.css (hunk 4 above)
Comment (glen-aot) [reply frontend/static_home_page/css/styles.css:114]: I have used the .h5 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/css/styles.css (hunk 5 above)
Comment (glen-aot) [reply frontend/static_home_page/css/styles.css:108]: This is the convention that was already in place when I took on this task, trying to stay consistent with the code already written

File: frontend/static_home_page/home.html (hunk 30 above)
Comment (glen-aot) [reply frontend/static_home_page/home.html:211]: I have used the .h5 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/home.html (hunk 31 above)
Comment (glen-aot) [reply frontend/static_home_page/home.html:121]: I have used the .h4 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/home.html (hunk 32 above)
Comment (glen-aot) [reply frontend/static_home_page/home.html:135]: I have used the .h4 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/home.html (hunk 33 above)
Comment (glen-aot) [reply frontend/static_home_page/home.html:160]: I have used the .h4 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/home.html (hunk 34 above)
Comment (glen-aot) [reply frontend/static_home_page/home.html:168]: I have used the .h4 class here to keep the style consistent with the design whilst keeping the heading levels in descending order in the markup

File: frontend/static_home_page/css/styles.css (hunk 9 above)
Comment (glen-aot) [reply frontend/static_home_page/css/styles.css:181]: This is the convention that was already in place when I took on this task, trying to stay consistent with the code already written

File: frontend/static_home_page/css/styles.css (hunk 3 above)
Comment (zgong-gov) [reply frontend/static_home_page/css/styles.css:51]: Keep the prefix for now, they can be changed later.

File: frontend/static_home_page/css/styles.css (hunk 4 above)
Comment (zgong-gov) [reply frontend/static_home_page/css/styles.css:114]: Use h5, h4, h3 elements then, and remove the class names. The ascending/descending order doesn't matter.

File: frontend/static_home_page/css/styles.css (hunk 5 above)
Comment (zgong-gov) [reply frontend/static_home_page/css/styles.css:108]: To be honest the convention in-place was supposed to be changed anyways, except the team never got to it. For the sake of completing this feature, I guess the class name conventions can stay as-is for the time being, but need to be refactored later on though.

File: frontend/static_home_page/home.html (hunk 30 above)
Comment (zgong-gov) [reply frontend/static_home_page/home.html:211]: Remove all .h5, .h4, and .h3 class names, and assign the styles associated with these classes to the h5, h4, h3 elements instead.

Comment (zgong-gov) [reply frontend/static_home_page/home.html:211]: Conceptually, if something is to be assigned a class name of ".h5", then it is an h5 element and can be expressed as such in HTML. Similarly for h4 and h3 as well.

File: frontend/static_home_page/home.html (hunk 31 above)
Comment (zgong-gov) [reply frontend/static_home_page/home.html:121]: See my reply above.

Comment (zgong-gov) [reply frontend/static_home_page/home.html:121]: Change the element to h4 element then, and remove the class name. Just ensure that the design styles match the ones assigned to the h4 in the css.

**Review by zgong-gov — APPROVED** [review]
Changes will be made in a future refactoring task.
